Makefile

```diff
@@ -134,6 +134,19 @@ check-scripts:
 
 test: all
 	@echo
+ifeq ($(TREAT_WARNINGS_AS_ERRORS), true)
```

Comment:
These makefile changes are really not very pretty. I suggest checking the variable directly in the utility command to avoid these.

Reply:
Do you know wether anything else should use this variable? Atm there's no precedence for checking environment variables outside of Program and SDL wrapper, the very lowest level of the engine

Reply:
Theres no strong precident for using flag arguments either (only the `--no-(shadow|padding)` on ConvertSpriteToPng IIRC).

Precident for using an environment variable include `MONO_LOG_LEVEL` for modifying warning/error/etc output, and Python has `PYTHONWARNINGS` that fits a similar role.

Reply:
I've changed it to fetch the environment variable in hopes this it's a generic enough variable to be reused